### ISSUE TITLE
[dev/xml] Fix debug enabled option in XslCompiledTransform

### ISSUE THREAD

**Issue by sepidehMS** [dotnet/corefx#10023]
Fix commented-out code (Marked with `//BinCompat TODO`) in XmlILModule supposed to run when XslCompiledTransform has enabled debug. 

When debug=true is passed to XslCompiledTransform, in the CompileQilToMsil step of the xsl transform the dynamic method is not being baked correctly (either due to the dynamic assembly not being generated correctly, or the dynamic module) and we get Object null reference error when the dynamic method Execute is being called. 

Also, re-enable tests tracking this issue.

**Comment by danmosemsft** [comment]
@sepidehMS Can you summarize which API are missing (and thus blocking enabling this)

**Comment by sepidehMS** [comment]
As this requires debugging IL generation, I am not 100% on what exactly is causing this break. But as far as my investigation goes, here are the some missing APIs that could be related:
```
System.Reflection.Emit.AssemblyBuilder.DefineDynamicModule(string, string)
System.Reflection.Emit.AssemblyBuilder.Save(string)
System.Reflection.Emit.DefineParameter(int, ParameterAttributes, string)
System.Reflection.Emit.ModuleBuilder.GetSymWriter()
```

**Comment by Petermarcu** [comment]
Is this in netstandard2.0 per the label? If so, please put it in the 1.2 milestone.

**Comment by karelz** [comment]
RefEmit is out of scope now.
